src/ucp/api/ucp.h

```diff
@@ -2357,6 +2377,37 @@ ucs_status_t ucp_mem_advise(ucp_context_h context, ucp_mem_h memh,
                             ucp_mem_advise_params_t *params);
 
 
+/**
+ * @ingroup UCP_MEM
+ * @brief Memory copy routine
+ *
+ * This routine performs memory copy from @a src to @a dest buffer, given the
+ * corresponding data and memory types specified in @a params. Refer to
+ * @ref ucp_request_param_t.target_datatype for @a dest data type and
+ *  @ref ucp_request_param_t.datatype for @a src data type.
+ *
+ * @param [in]  worker      Worker handle
+ * @param [in]  dest        Destination buffer to copy into
+ * @param [in]  dest_count  Number of elements of
+ *                          @ref ucp_request_param_t.target_datatype to copy from
+ *                          @a src buffer.
+ * @param [in]  src         Source buffer to copy from
+ * @param [in]  src_count   Number of elements of @ref ucp_request_param_t.datatype
+ *                          to copy into @a dest buffer.
+ * @param [in]  params      Operation parameters, see @ref ucp_request_param_t
+ *
+ * @return NULL                 - The operation was completed immediately
+ * @return UCS_PTR_IS_ERR(_ptr) - The operation failed
+ * @return otherwise            - Operation was scheduled and can be completed in
```

Comment:
`UCS_INPROGRESS`?
I guess `UCS_OK` is not returned from the function - correct me if I'm wrong

Reply:
it is `ucs_status_ptr_t` - request is returned in case of `IN_PROGRESS`